pad.c

```diff
@@ -1156,8 +1194,19 @@ S_pad_findlex(pTHX_ const char *namepv, STRLEN namelen, U32 flags, const CV* cv,
                     fake_offset = offset; /* in case we don't find a real one */
                     continue;
                 }
-                if (PadnameIN_SCOPE(name, seq))
+                if (!PadnameIN_SCOPE(name, seq))
+                    continue;
+
+                if (!PadnameIsTOMBSTONE(name))
                     break;
+
+                /* TODO(leonerd 2024-01-25): For better performance, we could
+                 * have performed this lookup at add time and just skip
+                 * immediately now, but there's currently no spare field in 
+                 * the padname struct to store the result.
+                 */
```

Comment:
😢 I think we've hit these issues before, where we have various internal structures in Perl which are exposed directly instead of hiding behind APIs, leading to things like this.

Reply:
do you mean opaque types?

Reply:
What if one bit in the padname indicates indirection (for the non-critical tasks) as in inode tables or UTF-8?
Or am I talking bullshit here?

Reply:
@Tux what about extending it with common (in C word in observed) generic structure `callback_data` + `callback_cleanup` ?

Instead of callback cleanup, `name` can be extended to:
```
int symbol_table;
char * name_without_sigil;
```

so `callback_cleanup` will be provided by `symbol_table` ? (I'm already working on `symbol_table` extension, though slowly)

Reply:
> or UTF-8?

padnames are always UTF-8.

I don't see any problems with extending the `padname` structure with extra fields, as long as we don't go wild.

The `xpadn_fieldinfo` could be turned into a union of the field info pointer and whatever other space we need if we want some "free" space unused for sub padnames.

Reply:
> I don't see any problems with extending the `padname` structure with extra fields, as long as we don't go wild.
> 
> The `xpadn_fieldinfo` could be turned into a union of the field info pointer and whatever other space we need if we want some "free" space unused for sub padnames.

Yeah that's probably going to be a fix I'll do at some point soon. I mostly didn't want the vast majority (~99%) of Padname structures to have to pay the memory cost of those unused fields.

Thinking more, at the time you create the Padname struct you always know if it's going to be for a field or a tombstone. Perhaps in that case it could actually allocate a "PadnameFull" structure with those extra fields; then the bitfllags indicate if those are valid or not. It means that normal (small) Padname structs don't need those fields, so we get some memory savings.